uniffi_meta/src/lib.rs

```diff
@@ -235,19 +238,24 @@ pub enum Type {
     ForeignExecutor,
     SystemTime,
     Enum {
+        crate_name: String,
         name: String,
     },
     Record {
+        crate_name: String,
```

Comment:
Changing `crate_name` to `module_path` makes sense to me.

I think it's reasonable to encode the crate_name/module path in the `Type` metadata is that metadata is shared between all crates.  Functions from one crate should see that type as a `Type::Record`, while functions from another crate should see it as a `Type::External`.

However, maybe a better system would be to lean into the concept of type identifiers and only store a path to the type definition.  I'm picturing something like:
  - There's a `TypeDefinition` enum with variants for record, object, etc.
  - There's a `TypeDefinitionUniverse` that maps full module paths, including the crate name, to `TypeDefinition`s
  - The `TYPE_RECORD`, `TYPE_ENUM`, etc. constants that are used for function signature arguments are replaced with `TYPE_USER`, which is followed by a module path string that keys into `TypeDefinitionUniverse`.
  - All of this can be used to construct a `ComponentInterface` for each crate

Side note: currently `module_path` is just a crate name, because of some reasons that I forgot.  I believe it should be possible to make it a full path though.